### PR TITLE
Added extra valhalla_run_isochrone functionality and fixed color bug in isochrone queries

### PR DESCRIPTION
**Pull request by acwilton**
# Changes
+ Added a "show_locations" parameter to the isochrone service (false by default). If set to true, the final geojson output will include all of the "locations" points as geojson points. That way (especially once multiple locations are allowed) you can see exactly where the isochrones are expanding from if you are visualizing the geojson output.
+ Fixed color issue with isochrones where colors would be placed in reverse order for contours and would be parsed incorrectly if only some contours specified colors while others didn't. Now if a contour has a color specified to it, that specific contour gets that color. All other contours with no color specified get a default color.
+ Added more functionality to valhalla_run_isochrone for better testing. It will now actually read json contour parameters (they were ignored before), and also the polygons, show_locations, generalize, and denoise parameters as described in the isochrone service api.
+ A command line argument has also been added for the polygons (-p), show_locations (-l), generalize (-g), and denoise (-d) parameters if you wanted to use those instead of in the json. (If both are used, a warning is displayed and the json overwrites the command line option)
+ The polygons default in valhalla_run_isochrone is now false to match the default value of polygons for the actual isochrone service.
+ Added a file command line parameter (-f) to valhalla_run_isochrone which will dump the geojson data into the specified file instead of onto the console.

### CODE REVIEW

File: src/baldr/geojson.cc [hunk 1]
```diff
@@ -19,14 +19,13 @@ template <class coord_t>
 MapPtr to_geojson(const typename midgard::GriddedData<coord_t>::contours_t& grid_contours, bool polygons, const std::vector<std::string>& colors) {
   //for each contour interval
   int i = 0;
-  auto color_itr = colors.cbegin();
+  auto color_itr = colors.crbegin();
```
Comment (kevinkreiser) [inline src/baldr/geojson.cc]: this doesnt seem right to me. it was broken before but this doesnt fix it.. i think

so the `grid_contours` are sorted from "largest" to "smallest", right? we need to take care to sort the `colors` vector accordingly. below (and in the service) it looks to me like we just use the `colors` in the order that they appear in the input which is to say there is no sorting enforced. what we need to do is sort the `colors` in the same order that `grid_contours` are sorted, ie by the time value.

Comment (acwilton) [inline src/baldr/geojson.cc]: Ya that makes sense. We will have to somehow associate the color with the time value which might be a bit complicated or just take some api fixes and stuff. But currently if the time values are out of order in your query, loki rejects the query completely. So my fix works, it just won't if we decide to release that restriction.

Comment (kevinkreiser) [inline src/baldr/geojson.cc]: just change the colors to a map of time to color and pass that instead right? then its not even an iterator its just a `color_map.find(interval.first)` and if its not in the map then we use the on the fly color derived from `i`. wont that work and be fool proof. sorry for not doing this the first go round...

Comment (acwilton) [inline src/baldr/geojson.cc]: Ok ya I like that idea a lot. I'll do that now.

**Pull request merged**